sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -39,6 +40,19 @@ namespace Azure { namespace Core {
      */
     std::string ToString();
 
+    /**
+     * @brief Construct a Uuid from an existing UUID represented as an array of bytes.
+     * @details Creates a Uuid from a UUID created in an external scope.
+     */
+    explicit Uuid(std::array<uint8_t, UuidSize> const& uuid) { m_uuid = uuid; }
+
+    /**
+     * @brief Returns the internal value of the Uuid for consumption by clients who need non-string
+     * representation of the Uuid
+     * @returns An array with the binary representation of the Uuid.
+     */
+    explicit operator std::array<uint8_t, UuidSize> const &() const { return m_uuid; }
```

Comment:
I think we should use a named method here (GetByteArray), rather than an operator overload.

Reply:
Personally, I don't think it's necessary, but it doesn't hurt to remove the constructor and operator overload (since IMHO this comment is objecting to the ability to directly transform a Uuid into an array of 16 bytes, removing both the constructor and operator overload make sense).